tools/conf-dtypes.txt

```diff
@@ -0,0 +1,63 @@
+[APY]
```

Comment:
While this file is OK, I'd much rather we not have it at all. It seems to be largely possible to automatically differentiate between these types.

Reply:
But there is a chance that a number is meant to be a string. Also, a flawky config file should not mis-determine the dtype of the value.

Reply:
Where should a number be interpreted as a string? That just sounds like a misleading configuration variable that should be replaced with an enum or something.

Reply:
The line  `CONSTANT_NAME = 42` would be "automatically differentiated" to `int` even if `CONSTANT_NAME` is a `string` and should be `"42"`

Reply:
I don't deny that it's theoretically possible. Rephrased: Where should a number be interpreted as a string [in a reasonable setting with our set of options or a reasonable extension]?

Reply:
The balance to be stricken here is between being strict and type safe or being more maintainable and having less repetition. Unless there's a good argument for the former, the latter is preferable.

Reply:
@shardulc initially wanted a dict in `read-conf.py` but I suggested to have a separate file for that.

Reply:
I prefer a dict in `read-conf.py` over this as well and over both I prefer not having a list at all. The example configuration files should serve as documentation for the options. If one is unclear, a comment would be allay the concern.

Reply:
The separate file also introduces a new syntax which is not ideal.

Reply:
Consider for example how Django does their `settings.py`. There's no exhaustive list of options.